vsintegration/src/FSharp.Editor/CodeFix/ConvertToAnonymousRecord.fs

```diff
@@ -3,45 +3,49 @@
 namespace Microsoft.VisualStudio.FSharp.Editor
 
 open System.Composition
-open System.Threading
 open System.Threading.Tasks
 open System.Collections.Immutable
 
 open Microsoft.CodeAnalysis.Text
 open Microsoft.CodeAnalysis.CodeFixes
-open Microsoft.CodeAnalysis.CodeActions
+
+open CancellableTasks
 
 [<ExportCodeFixProvider(FSharpConstants.FSharpLanguageName, Name = CodeFix.ConvertToAnonymousRecord); Shared>]
 type internal FSharpConvertToAnonymousRecordCodeFixProvider [<ImportingConstructor>] () =
     inherit CodeFixProvider()
 
-    static let title = SR.ConvertToAnonymousRecord()
-
     override _.FixableDiagnosticIds = ImmutableArray.Create("FS0039")
 
-    override _.RegisterCodeFixesAsync context : Task =
-        asyncMaybe {
-            let document = context.Document
+    interface IFSharpCodeFix with
+        member _.GetChangesAsync document span =
+            cancellableTask {
+                let! cancellationToken = CancellableTask.getCurrentCancellationToken ()
```

Comment:
Would be better to make the correct cancellation token (the one from Roslyn) a direct argument of the function.
Firstly to make sure the right one is used for sure, secondly to avoid repetition since every non-trivial codefix will need anyway (just to get the document text for example).

Reply:
Hmm point taken - but, let's see and address a few more code fixes. At a glance, indeed, majority of the code fixes need a cancellation token, yet some don't. Some don't even need a document at all, the span is enough for the code fix logic itself. 

I will be trying to find a balance for this "one-fits-all" code fix interface but I might also not come up with a good one and we'll see what to do then :)

Reply:
Also, it is already explicit there, return type makes it.